src/DIRAC/FrameworkSystem/API/AuthHandler.py

```diff
@@ -435,9 +435,9 @@ def __researchDIRACGroup(self, extSession, chooseScope, state):
         username = extSession["authed"]["username"]
         # Requested arguments in first request
         provider = firstRequest.provider
-        self.log.debug(f"Next groups has been found for {username}:", ", ".join(firstRequest.groups))
+        self.log.debug("The following groups found", f"for {username}: {', '.join(firstRequest.groups)}")
 
-        # Researche Group
+        # Look for DIRAC groups valid for the user
```

Comment:
I still don't understand if there should be some comparison between the result of `getGroupsForUsers` and `firstRequest.groups`
and if not why not do this 
```python
        # If group already defined in first request, just return it
        if firstRequest.groups:
          return extSession["authed"], firstRequest
```
before `getGroupsForUsers`

Reply:
I have moved these lines upstream because the validity of the group specified by a user initially is already checked by this point. In any case, I would like to review the logic of the requested group validation but not as part of this PR. I have checked that invalid groups specified in dirac-login do not pass, this is enough for the moment.

Reply:
Yes, thanks!